scalyr_agent/test_util.py

```diff
@@ -183,10 +186,13 @@ def parse_scalyr_request(payload):
         # First add in the bytes between the last processed and the start of this match.
         rewritten_payload += payload[last_processed_index + 1 : x.start(0)]
         # Read the 4 bytes that describe the length, which is stored in regex group 1.
-        length = struct.unpack(">i", x.group(1))[0]
+        # 2->TODO struct.pack|unpack in python2.6 does not allow unicode format string.
+        length = struct.unpack(six.ensure_str(">i"), x.group(1))[0]
         # Grab the string content as raw bytes.
         raw_string = payload[x.end(1) : x.end(1) + length]
-        rewritten_payload += scalyr_util.json_encode(raw_string.decode("utf-8"))
+        rewritten_payload += six.ensure_binary(
+            scalyr_util.json_encode(raw_string.decode("utf-8"))
```

Comment:
Actually, there are a lot more places where we need a text string from `scalyr_util.json_encode` rather than binary string.
Considering this and the fact that python3 json libraries return only text strings, always return binary strings seems like extra work. 
Please take a look at my current variant. I've just introduced an additional keyword argument - `binary` and passing it when binary string is needed.

Reply:
Looked this over, this is ok.